taskcluster/kinds/compile/kind.yml

```diff
@@ -0,0 +1,84 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+---
+
+loader: taskgraph.loader.transform:loader
+
+transforms:
+    - translations_taskgraph.transforms.worker_selection
+    - taskgraph.transforms.task_context
+    - taskgraph.transforms.run:transforms
+    - translations_taskgraph.transforms.cached_tasks:transforms
+    - taskgraph.transforms.task:transforms
+
+kind-dependencies:
+    - toolchain
+
+tasks:
+    marian-fork:
+        description: >
+            Compile the marian-fork inference engine for use in the pipeline. This contains
+            the --int8shiftAlphaAll quantization, which is used in our production models.
+        attributes:
+            stage: compile
+            cache:
+                type: compile
+                # Since inference/marian-fork is a large subfolder, we don't want to
+                # invalidate the prod builds for any small change in the marian-fork.
+                # Instead, rely on a version number that can be bumped when it needs
+                # to be rebuilt.
+                version: 2
+                resources:
+                    - inference/scripts/build.py
+                    # This matches recursively, but only invalidates the task in CI runs.
+                    - inference/marian-fork/src/command/*
```

Comment:
Should this simply be `inference/marian-fork` instead? (I wonder if you're doing this to workaround an upstream bug, similar to what was recently fixed in https://github.com/taskcluster/taskgraph/pull/723?)